android/src/main/java/com/heapanalytics/reactnative/RNHeapLibraryModule.java

```
@@ -0,0 +1,110 @@
+  }
+
+  @ReactMethod
+  public void setAppId(String appId) {
+    try {
+      Field skipInstrumentorCheckField = HeapImpl.class.getDeclaredField("skipInstrumentorChecks");
+      skipInstrumentorCheckField.setAccessible(true);
+      skipInstrumentorCheckField.setBoolean(null, true);
+    } catch (NoSuchFieldException e) {
+      System.out.println("Caught NoSuchFieldException when trying to skip the instrumentor checks");
+      e.printStackTrace();
+    } catch (IllegalAccessException e) {
+      System.out.println("Caught IllegalAccessException when trying to skip the instrumentor checks");
+      e.printStackTrace();
+    }
+
+    Heap.init(this.reactContext, appId);
+  }
+
+  @ReactMethod
+  public void identify(String identity) {
+    Heap.identify(identity);
+  }
+
+  private static Map<String, String> convertToStringMap(ReadableMap readableMap) {
+    Map<String, String> stringMap = new HashMap<>();
+    ReadableMapKeySetIterator mapIterator = readableMap.keySetIterator();
+    while (mapIterator.hasNextKey()) {
+      String key = mapIterator.nextKey();
+      switch (readableMap.getType(key)) {
+        case Null:
+          stringMap.put(key, null);
+          break;
+        case Number:
+          stringMap.put(key, "" + readableMap.getDouble(key));
+          break;
+        case Boolean:
+          stringMap.put(key, readableMap.getBoolean(key) ? "true" : "false");
+          break;
+        case String:
+          stringMap.put(key, readableMap.getString(key));
+          break;
+        case Array:
+          List<Object> list = readableMap.getArray(key).toArrayList();
+          List<String> stringList = new ArrayList<>();
+          for (Object o : list) {
+            stringList.add(o.toString());
+          }
+          stringMap.put(key, TextUtils.join(",", stringList));
+          break;
+        case Map:
+          // In many cases, this won't get used, since we flatten incoming
+          // objects before they get to this point.
```

whose error is it if this happens? if it's our responsibility to flatten, and this shouldn't happen at all, we should throw InvalidArgumentException or at least log.

---

also could you document requirements on the input to this method up top?